Logs failed SA.UpdatePendingAuthorization.

cpu opened the pull request:
This commit resolves #2303 by updating the comment, and returned error type produced when the RA calls `SA.UpdatePendingAuthorization` and it fails.

Previously this produced a `MalformedRequestError` that was described as only happening when the client corrupted the challenge data.

Now this is returned as more descriptive `ServerInternalError` and the underlying error from the SA is logged as a warning for further debugging.

rolandshoemaker commented on ra/ra.go:
```diff
@@ -963,9 +963,14 @@ func (ra *RegistrationAuthorityImpl) UpdateAuthorization(ctx context.Context, ba
-		// This can pretty much only happen when the client corrupts the Challenge
-		// data.
-		err = core.MalformedRequestError("Challenge data was corrupted")
+		//Previously this was a MalformedREquestError alongside a comment explaning
```
Could you add spaces between the `//` and text.

cpu replied:
Good catch! Yes, will change. Thanks

jsha commented on ra/ra.go:
```diff
@@ -963,9 +963,14 @@ func (ra *RegistrationAuthorityImpl) UpdateAuthorization(ctx context.Context, ba
-		// This can pretty much only happen when the client corrupts the Challenge
-		// data.
-		err = core.MalformedRequestError("Challenge data was corrupted")
+		// Previously this was a MalformedREquestError alongside a comment explaning
```
While I'm a big fan of commenting in general, I think we don't need a comment documenting what used to be here, just what is here. And since logging an error is the straightforward, normal thing to do, I don't think this specifically needs a comment.

I'd also change the `%q` in the log to a `%s`, which is how we typically do error interpolation.

cpu replied:
:+1: Makes sense - the history is there in git for the curious archaeologist and it isn't very interesting.

cpu commented:
@jsha addressed your feedback w/ https://github.com/letsencrypt/boulder/pull/2321/commits/e7d8922fe284dc891d302e00fb71a39b0edaf76f. Ready for another :eyes: - thanks!
